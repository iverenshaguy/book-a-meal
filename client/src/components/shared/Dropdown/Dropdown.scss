@import '../../../../public/scss/_variables';
.dropdown {
  position: relative;
  display: inline-block;
}

.navbar .dropdown {
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.dropdown .dropdown-content {
  display: none;
  flex-direction: column;
  position: absolute;
  top: 100%;
  border-radius: 3px;
  min-width: 120px;
  text-align: center;
  border: 1px solid #ddd;
  background-color: white;
  box-shadow: 4px 4px 8px 0 rgba(0, 0, 0, 0.05);
  right: 0%;
}

.navbar .dropdown-content {
  top: 100%;
  right: -20%;
}

.admin .navbar .admin-dropdown-content {
  top: 80%;
}

.dropdown {
  &.notification .dropdown-content {
    width: 300px;
    max-height: 70vh;
    overflow: auto;
    z-index: 9;
  }
  .dropdown-content {
    &.show {
      display: flex;
    }
    a,
    .link-btn {
      color: $dark;
      padding: 10px;
      font-family: $heading-font-stack;
      font-display: swap;
      border-bottom: 1px solid #ddd;
      &:last-child {
        border-bottom: 0 solid #ddd;
      }
      &:hover,
      &:focus {
        color: $primary-color;
        background-color: $lightgrey;
        transition: color 0.5s, background-color 0.5s;
      }
    }
    &.profile {
      margin: 0;
    }
  }
  &.notification .dropdown-content a,
  &.notification .dropdown-content .link-btn {
    padding: 20px 15px;
    &.read {
      background-color: $lightgrey;
    }
  }
}

@media (max-width: 767.98px) {
  .admin .navbar .admin-dropdown-content {
    top: 100%;
  }
}

@media (max-width: 470px) {
  .navbar .dropdown.notification .dropdown-content {
    right: 0%;
  }
}