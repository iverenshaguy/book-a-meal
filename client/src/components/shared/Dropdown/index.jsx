import React, { Component } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import LinkBtn from '../Link';
import './Dropdown.scss';

/**
 * @exports
 * @class Dropdown
 * @extends Component
 * @classdesc Creates Dropdown Component
 * @returns {JSX} Dropdown Component
 */
class Dropdown extends Component {
  /**
   * @constructor
   * @returns {JSX} Dropdown
   */
  constructor() {
    super();

    this.state = {
      showContent: false
    };
  }

  /**
   * @memberof Dropdown
   * @returns {void}
   */
  toggleDropdownContent = () => {
    this.setState(({ showContent }) => ({
      showContent: !showContent
    }));
  }

  /**
   * @memberof Dropdown
   * @returns {JSX} Dropdown Component
   */
  render() {
    const dropdownClass = classNames({
      dropdown: true,
      notification: this.props.type === 'notification' || this.props.type === 'admin-notification',
      'card-dropdown': this.props.type === 'card'
    });

    const dropdownContentClass = classNames({
      show: this.state.showContent,
      'dropdown-content': true,
      'admin-dropdown-content': this.props.type === 'admin-notification',
      meal: this.props.type === 'card'
    });

    return (
      <div
        className={dropdownClass}
        onMouseEnter={this.toggleDropdownContent}
        onMouseLeave={this.toggleDropdownContent}
      >
        <LinkBtn id="dropdown-toggler" className="link">
          {this.props.toggler}
        </LinkBtn>
        <div className={dropdownContentClass} id="dropdown-content">{this.props.content}</div>
      </div>
    );
  }
}

Dropdown.propTypes = {
  type: PropTypes.string.isRequired,
  toggler: PropTypes.element.isRequired,
  content: PropTypes.element.isRequired,
};

export default Dropdown;
