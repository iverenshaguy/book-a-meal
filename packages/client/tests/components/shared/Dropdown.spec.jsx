import React from 'react';
import Dropdown from '../../../src/components/shared/Dropdown';

describe('Dropdown', () => {
  it('should render the Dropdown component correctly when the type prop is notification', () => {
    const shallowWrapper = shallow(<Dropdown
      type="notification"
      toggler={<p>Click Me</p>}
      content={<div>Hey</div>}
    />);


    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('should render the Dropdown component correctly when the type prop is admin-notification', () => {
    const shallowWrapper = shallow(<Dropdown
      type="admin-notification"
      toggler={<p>Click Me</p>}
      content={<div>Hey</div>}
    />);


    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('should toggle the showContent state to true when toggle is hovered on', () => {
    const wrapper = mount(<Dropdown
      type="notification"
      toggler={<p>Hover over Me</p>}
      content={<div>Hey</div>}
    />);

    wrapper.find('.dropdown').simulate('mouseEnter');
    expect(wrapper.state('showContent')).toEqual(true);

    wrapper.find('.dropdown').simulate('mouseLeave');
    expect(wrapper.state('showContent')).toEqual(false);

    wrapper.unmount();
  });
});
