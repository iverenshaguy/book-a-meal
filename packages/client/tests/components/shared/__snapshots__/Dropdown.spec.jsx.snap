// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render the Dropdown component correctly when the type prop is admin-notification 1`] = `
<div
  className="dropdown notification"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <Link
    className="link"
    clickHandler={null}
    dataTip={null}
    id="dropdown-toggler"
  >
    <p>
      Click Me
    </p>
  </Link>
  <div
    className="dropdown-content admin-dropdown-content"
    id="dropdown-content"
  >
    <div>
      Hey
    </div>
  </div>
</div>
`;

exports[`Dropdown should render the Dropdown component correctly when the type prop is notification 1`] = `
<div
  className="dropdown notification"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <Link
    className="link"
    clickHandler={null}
    dataTip={null}
    id="dropdown-toggler"
  >
    <p>
      Click Me
    </p>
  </Link>
  <div
    className="dropdown-content"
    id="dropdown-content"
  >
    <div>
      Hey
    </div>
  </div>
</div>
`;
